File: mindspore/lite/minddata/example/testlenet.cpp
/**
 * Copyright 2021  Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
#include <sys/stat.h>
#include <unistd.h>
#include <map>
#include <memory>
#include <set>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <utility>
#include <vector>

#include "include/datasets.h"
#include "include/iterator.h"
#include "include/vision_lite.h"
#include "include/transforms.h"
#include "include/api/types.h"

using mindspore::dataset::Dataset;
using mindspore::dataset::Iterator;
using mindspore::dataset::Mnist;
using mindspore::dataset::TensorTransform;

int main(int argc, char **argv) {
  std::string folder_path = "./testMnistData/";
  std::shared_ptr<Dataset> ds = Mnist(folder_path, "all");

  std::shared_ptr<TensorTransform> resize(new mindspore::dataset::vision::Resize({32, 32}));
  ds = ds->Map({resize});

  ds = ds->Shuffle(2);
  ds = ds->Batch(2);

  std::shared_ptr<Iterator> iter = ds->CreateIterator();

  std::unordered_map<std::string, mindspore::MSTensor> row;
  iter->GetNextRow(&row);

  uint64_t i = 0;
  while (row.size() != 0) {
    i++;
    iter->GetNextRow(&row);
  }

  iter->Stop();
}

File: tests/perf_test/test_lenet.py
# Copyright 2020 Huawei Technologies Co., Ltd
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# ============================================================================

"""LeNet test."""

import numpy as np

from lenet import LeNet5
import mindspore.nn as nn
import mindspore.ops.composite as C
from mindspore import Tensor
from mindspore import context
from mindspore.common.api import _cell_graph_executor

context.set_context(mode=context.GRAPH_MODE)


grad_all_with_sens = C.GradOperation(get_all=True, sens_param=True)

batch_size = 1
channel = 1
height = 32
weight = 32
num_class = 10


class LeNetGrad(nn.Cell):
    """Backward of LeNet"""

    def __init__(self, network):
        super(LeNetGrad, self).__init__()
        self.grad_op = grad_all_with_sens
        self.network = network

    def construct(self, x, sens):
        grad_op = self.grad_op(self.network)(x, sens)

        return grad_op


def test_compile():
    """Compile forward graph"""
    net = LeNet(num_class=num_class)
    np.random.seed(7)
    inp = Tensor(np.array(np.random.randn(batch_size,
                                          channel,
                                          height,
                                          weight) * 3, np.float32))

    _cell_graph_executor.compile(net, inp)


def test_compile_grad():
    """Compile forward and backward graph"""
    net = LeNet5(num_class=num_class)
    inp = Tensor(np.array(np.random.randn(batch_size,
                                          channel,
                                          height,
                                          weight) * 3, np.float32))
    sens = Tensor(np.ones([batch_size, num_class]).astype(np.float32))
    grad_op = LeNetGrad(net)

    _cell_graph_executor.compile(grad_op, inp, sens)
